packages/x-date-pickers/src/LocalizationProvider/LocalizationProvider.tsx

```diff
@@ -52,31 +52,25 @@ export interface LocalizationProviderProps<TDate> {
  * @ignore - do not document.
  */
 export function LocalizationProvider<TDate>(inProps: LocalizationProviderProps<TDate>) {
+  const { localeText: inLocaleText, ...otherInProps } = inProps;
+
   const { utils: parentUtils, localeText: parentLocaleText } = React.useContext(
     MuiPickersAdapterContext,
   ) ?? { utils: undefined, localeText: undefined };
 
-  const tempLocaleText = React.useMemo(
-    () => ({ ...parentLocaleText, ...inProps.localeText }),
-    [parentLocaleText, inProps.localeText],
-  );
-
-  const props = useThemeProps({
-    props: {
-      ...inProps,
-      localeText: tempLocaleText,
-    },
+  const props: LocalizationProviderProps<TDate> = useThemeProps({
+    // We don't want to pass the `localeText` prop to the theme, that way it will always return the theme value,
+    // We will then merge this theme value with our value manually
+    props: otherInProps,
     name: 'MuiLocalizationProvider',
   });
 
-  const {
-    children,
-    dateAdapter: DateAdapter,
-    dateFormats,
-    dateLibInstance,
-    adapterLocale,
-    localeText,
-  } = props;
+  const localeText = React.useMemo(
+    () => ({ ...props.localeText, ...parentLocaleText, ...inLocaleText }),
+    [props.localeText, parentLocaleText, inLocaleText],
+  );
```

Comment:
`props.localeText` is not obvious to get

What about moving this after `props` destructuring  to be able to write
```
() => ({ ...defaultizedLocaleText, ...parentLocaleText, ...inLocaleText }),
```

Reply:
OK for me, I'll just rename `defaultizedLocaleText` into `themeLocaleText` to make clear that it can only contain the theme value and never the value from the `inProps`